File: elixir/bob/bob.exs
defmodule Bob do
  defp is_question(msg), do: String.ends_with?(msg, "?")

  defp is_yelling(msg) do
    String.match?(msg, ~r/[^\d\W]/u) and msg == String.upcase(msg)
  end

  defp is_yelling_question(msg) do
    is_yelling(msg) and is_question(msg)
  end

  def hey(input) do
    cond do
      String.trim(input) == "" ->
        "Fine. Be that way!"
      is_yelling_question(input) ->
        "Calm down, I know what I'm doing!"
      is_yelling(input) ->
        "Whoa, chill out!"
      is_question(input) ->
        "Sure."
      true ->
        "Whatever."
    end
  end
end

File: python/bob/bob.py
class Teenager(object):
    brain = [
        (lambda msg: not msg, 'Fine. Be that way!'),
        (lambda msg: msg.isupper() and msg.endswith('?'), 'Calm down, I know what I\'m doing!'),
        (lambda msg: msg.isupper(), 'Whoa, chill out!'),
        (lambda msg: msg.endswith('?'), 'Sure.'),
        (lambda msg: True, 'Whatever.')
    ]

    def __init__(self, name: str) -> None:
        self.name = name

    def talk(self, something: str) -> str:
        message = something.strip()
        for func, resp in self.brain:
            if func(message):
                return resp

def hey(phrase):
    return Teenager('Bob').talk(phrase)
